### PR TITLE
chore(react-dialog): Adds e2e tests

### PR DESCRIPTION
**Pull request by bsunderhus**

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `190.54 kB`<br />`52.487 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `32.036 kB`<br />`10.577 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                                    | `83.895 kB`<br />`25.077 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/acb25cb4eb784867ac87de03ee4efa08dedbc391'>acb25cb4eb784867ac87de03ee4efa08dedbc391</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/Avatar-mount_base.html">1611</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/Avatar-mount.html">1627</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/Button-mount_base.html">1090</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/Button-mount.html">1078</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/FluentProvider-mount_base.html">1918</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/FluentProvider-mount.html">1894</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/FluentProviderWithTheme-mount_base.html">691</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/FluentProviderWithTheme-mount.html">677</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">661</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">641</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">682</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">703</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/MakeStyles-mount_base.html">2242</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/MakeStyles-mount.html">2236</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/SpinButton-mount_base.html">3255</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24167/perf-test-react-components/SpinButton-mount.html">3233</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-dialog/e2e/Dialog.e2e.tsx [hunk 1]
```diff
@@ -4,19 +4,224 @@ import { mount as mountBase } from '@cypress/react';
...
+            <DialogTrigger>
+              <Button appearance="secondary">Close</Button>
+            </DialogTrigger>
+            <Button appearance="primary">Do Something</Button>
+          </DialogActions>
+        </DialogSurface>
+      </Dialog>,
+    );
+    cy.get('#open-btn').click();
+    cy.get('#dialog-surface').should('exist');
```
Comment (ling1726) [inline packages/react-components/react-dialog/e2e/Dialog.e2e.tsx:61]: nit: might want to hoist all these selectors, they are used quite a few times

Comment (ling1726) [reply packages/react-components/react-dialog/e2e/Dialog.e2e.tsx:61]: I would also use more specific selectors to the component like `[aria-haspopup]` which represent the actual DOM markup

**Review by ling1726 — APPROVED** [review]

**Pull request merged**